### DISCUSSION TITLE
Installing multiple architectures of the same extension in a single vscode instance?

### DISCUSSION THREAD

**Discussion by Lassebq** [microsoft/vscode-cpptools#12166] [Q&A] [answered]
I'm currently running Alpine Linux alongside a glibc-based distro. I'm also sharing a /home partition between these distros, which is where all my extensions get installed.

There are multiple downloads available for the C/C++ extension. When I install "Alpine Linux 64 bit", previously installed "Linux x64" gets overwritten and when I reboot back to the glibc-based distro I get "The C/C++ extension installed does not match your system" upon workspace initialization with a C project. Reinstalling "Linux x64" one fixes it but breaks extension for Alpine again.

Is there any way to keep both installed without having to manually reinstall/switch between extension architectures?

![image](https://github.com/microsoft/vscode-cpptools/assets/68742864/0f8d0f98-2089-4e0a-8b1b-5ca855aa1713)

**Comment by heartacker** [comment]
i have same require

**Comment by bobbrow** [comment]
An option that will work would be to install both regular VS Code and VS Code Insiders.  They both have their own extension caches.  So you could, for example, dedicate VS Code to standard linux and VS Code Insiders to Alpine and install the corresponding version of the C++ extension on each instance of VS Code.

Outside of this, I think you'd have to ask the VS Code team for a feature that would allow you to switch between the two versions in the same instance of VS Code.

**Reply by bobbrow** [reply]
I also just found a command switch that you could use with a single instance of VS Code.  `--extensions-dir` appears to allow you to set the location of the extension cache, so you could have separate locations for Alpine vs. standard linux, but it would require you to always run one of the distros from the command line so you could select the extension directory.  e.g. `code --extensions-dir ~/.vscode-alpine`

**Reply by Lassebq** [reply, marked as answer]
It's probably best if the extension supported both musl and glibc within the same release. From what I could find the "incompatiblity" check is implemented inside of extension.
Also I think a --bind mount would work better in my case. You don't need to run anything when you switch between distributions, you just make a mount point to another directory.
